workflows/argo-events/workflowtemplates/enroll-server.yaml

```diff
@@ -0,0 +1,45 @@
+---
+apiVersion: argoproj.io/v1alpha1
+metadata:
+  name: enroll-server
+  annotations:
+    workflows.argoproj.io/title: Perform server discovery and update Nautobot and Ironic
+    workflows.argoproj.io/description: |
+      Defined in `workflows/argo-events/workflowtemplates/enroll-server.yaml`
+kind: WorkflowTemplate
+spec:
+  arguments:
+    parameters:
+      - name: ip_address
+  templates:
+    - name: enroll-server
```

Comment:
So they'll be more steps that we'll want to run so really this should either be nested as part of steps and with an entry point or as a dag. An example of what we'll want to run next is in #415